src/main/java/com/dumbdogdiner/stickyapi/common/webapis/CachedMojangAPI.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2020 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.common.webapis;
+
+import com.dumbdogdiner.stickyapi.common.user.StickyUser;
+import com.dumbdogdiner.stickyapi.common.util.StringUtil;
+import com.dumbdogdiner.stickyapi.common.util.textures.DefaultSkins;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Player;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.UUID;
+
+//TODO: Better error handeling in case of 404
```

Comment:
Please do this.

Reply:
It's planned very shortly, I needed a review first